tests/test_on_data/test_gin_ecephys.py

```diff
@@ -448,7 +456,11 @@ class TestConverter(NWBConverter):
             # NWBSortingExtractor on spikeinterface does not yet support loading data written from multiple segment.
             if sorting.get_num_segments() == 1:
                 nwb_sorting = NwbSortingExtractorSI(file_path=nwbfile_path, sampling_frequency=sf)
-                check_sorting_equal_si(SX1=sorting, SX2=nwb_sorting)
+                # In the NWBSortingExtractor, since unit_names could be not unique,
+                # table "ids" are loaded as unit_ids. Here we rename the original sorting accordingly
+                sorting_renamed = sorting.select_units(unit_ids=sorting.unit_ids,
+                                                       renamed_unit_ids=np.arange(len(sorting.unit_ids)))
```

Comment:
this is a tricky one. The `spikeinterface.extractors.NWBSortingExtractor` used `id` as `unit_ids`. This is now required because `unit_name` might not be unique (we need to propagate the same change to recording extractors too). Since the original recording might have `str` unit_ids, we have to rename the original unit is to match the `nwb_sorting` object. The `check_*` function also checks for spike trains, so we are sure they are actually the same unit.